ethereum/core/src/main/java/org/hyperledger/besu/ethereum/worldstate/Pruner.java

```diff
@@ -70,27 +68,27 @@ public Pruner(
       final MarkSweepPruner pruningStrategy,
       final Blockchain blockchain,
       final PrunerConfiguration prunerConfiguration) {
-    this(pruningStrategy, blockchain, prunerConfiguration, getDefaultExecutorSupplier());
-  }
-
-  private static Supplier<ExecutorService> getDefaultExecutorSupplier() {
-    return () ->
+    this(
+        pruningStrategy,
+        blockchain,
+        prunerConfiguration,
         Executors.newSingleThreadExecutor(
             new ThreadFactoryBuilder()
                 .setDaemon(true)
                 .setPriority(Thread.MIN_PRIORITY)
                 .setNameFormat("StatePruning-%d")
-                .build());
+                .build()));
   }
 
   public void start() {
-
-    if (state.compareAndSet(State.IDLE, State.RUNNING)) {
-      LOG.info("Starting Pruner.");
-      executorService = executorServiceSupplier.get();
-      pruningStrategy.prepare();
-      blockAddedObserverId = blockchain.observeBlockAdded(this::handleNewBlock);
-    }
+    execute(
```

Comment:
Is this not going to create timing issues?  I think we may be intentionally blocking here, so that our listeners get setup before we start syncing and pulling in data.  Changing this to async, it seems like we might run into problems where:  we start importing blocks before our listeners are set up, we don't mark those initial state data, so we end up pruning useful state.

I'd suggest returning a future, and waiting for the future to complete before starting the sync task.

Reply:
I thought about this but it's prepare that sets up the block added listener so I think we're good.

Reply:
Ok, I think that makes sense - we start listening for world state, then start listening for blocks, and the block listener is what starts up the marking phase. 👍